instrumentation/rack/test/opentelemetry/instrumentation/rack/middlewares/tracer_middleware_test.rb

```diff
@@ -84,6 +84,91 @@
       end
     end
 
+    describe 'config[:request_hook] / config[:response_hook]' do
+      let(:query_string_attribute) { 'query_string' }
+      let(:response_body_attribute) { 'response_body' }
+      let(:query_string) { 'abcd=1234' }
+
+      describe 'when hooks are valid' do
+        let(:config) do
+          {
+            request_hook: lambda do |span, request|
+              span.set_attribute(query_string_attribute, request['QUERY_STRING'])
+            end,
+            response_hook: lambda do |span, response|
+              span.set_attribute(response_body_attribute, response[0])
+            end
```

Comment:
I'm not really sure the rack response body is something that's appropriate to expose as part of configuration, imo, especially given the lack of tests around how this behaves compared to the detail of the specification https://github.com/rack/rack/blob/main/SPEC.rdoc#the-body- . Reading the specification for it, there seems to be a number of untested code paths currently. Asking users to try to put in handling for all of these within their lambda seems unrealistic, and I tend to agree with @ahayworth that we may want to address specific use cases with configuration that we have confidence in.